recipes/apr/all/conanfile.py

```diff
@@ -120,7 +120,7 @@ def package(self):
 
     def package_info(self):
         self.cpp_info.names["pkg_config"] = "apr-1"
-        self.cpp_info.libs = ["apr-1"]
+        self.cpp_info.libs = ["libapr-1" if self.settings.os == "Windows" and self.options.shared else "apr-1"]
```

Comment:
```suggestion
        self.cpp_info.libs = ["libapr-1" if self.options.shared else "apr-1"]
```

Looking at upstream CMakeLists, there is no condition based on Windows.

Reply:
The package is built using CMake only when `compiler== "Visual Studio"`

https://github.com/conan-io/conan-center-index/blob/19302b69bedeaae3c207769fe605304e9002d399/recipes/apr/all/conanfile.py#L91-L98

Maybe I should change "Windows" for "Visual Studio". I left it as `self.settings.os == "Windows"` just to follow the same style as apr-util.

If that's incorrect then yeah, I can drop that condition

Reply:
Ok, missed that, so yes replace windows by visual studio please.

Reply:
Done.

Reply:
FYI, apr-util also used `self.settings.os == "Windows"`, so with this change they're still different, which is a little weird.

https://github.com/conan-io/conan-center-index/blob/bd0259673b8c6650f0f0fa96e5499d1983f84a1f/recipes/apr-util/all/conanfile.py#L184
https://github.com/conan-io/conan-center-index/blob/bd0259673b8c6650f0f0fa96e5499d1983f84a1f/recipes/apr/all/conanfile.py#L123

It's not really a problem because [conan-center-index only uses Visual Studio on Windows](https://github.com/conan-io/conan-center-index/blob/db24bf00520f63115af3294929bde03bdc0966d4/docs/supported_platforms_and_configurations.md#windows). Therefore right now either condition gives the same answers. I guess if or when someone updates the recipe to support windows clang or MinGW, they'd have to decide whether to use the CMake or autotools, and determine what to do. It's not just the CMakeLists.txt that uses this naming, the upstream apr-1.7.0 Makefile.win (also MSVC, but not used by this recipe) also uses the libapr-1.lib/apr-1.lib names.

But I do think `os==Windows` was logical, since the real driver is that you don't link to the .so vs .a like on ELF toolchain, whereas on PE you need to link with an implib that is generally the same format/file extension as a static library (.lib for MSVC, .a for gcc). So any other windows toolchains would have the same problem and have to solve it somehow; following the MSVC pattern seems most likely.